Utilities/MCStepLogger/include/MCStepLogger/AnalysisFileHandler.h

```diff
@@ -0,0 +1,73 @@
+/* Bookkeeping of AnalysisFiles objects
+ * Can be used when multiple files should be handled
+ */
+#ifndef ANALYSIS_FILE_HANDLER_H_
+#define ANALYSIS_FILE_HANDLER_H_
+
+#include <string>
+#include <vector>
+
+#include "TH1.h"
+
+namespace o2
+{
+namespace mcstepanalysis
+{
+
+class AnalysisFile;
+
+class AnalysisFileHandler
+{
+  // add as friend so it can access protected (private) members in order to use this class
+  friend class AnalysisFile;
+
+ public:
+  AnalysisFileHandler();
+  virtual ~AnalysisFileHandler() = default;
+  /// add file directly
+  void addAnalysisFile(AnalysisFile* analysisFile);
+  /// add a file from disk
+  void addAnalysisFile(const std::string& filepath);
+
+  /// check number of present analysis files for this meta analysis
+  size_t nAnalysisFiles() const;
+  size_t nAnalysisFiles(const std::string& analysisName) const;
+
+  /// get histogram by name and number since also overlaying should be enabled
+  template <typename T = TH1>
+  T* getHistogram(const std::string& analysisName, const std::string& histogramName, int level = 0)
+  {
+    static_assert(std::is_base_of<TH1, T>::value, "the object you want to get does not derive from TH1");
+    TH1* histo = findHistogram(analysisName, histogramName, level);
+    if (histo) {
+      return dynamic_cast<T*>(histo);
+    }
+    return nullptr;
+  }
+  /// get the meta of an analysis
+  const std::string& label(const std::string& analysisName, int level) const;
+  /// save files
+  void save(const std::string& outputDir) const;
+
+ private:
+  /// don't allow copy etc.
+  AnalysisFileHandler(const AnalysisFileHandler&);
```

Comment:
`C++11` would be `AnalysisFileHandler(const AnalysisFileHandler&) = delete;`

Reply:
although your comment is correct, this class is gone now anyway